sycl/include/CL/sycl/detail/pi.h

```diff
@@ -928,6 +929,142 @@ pi_result piEnqueueMemUnmap(
+  PI_MEM_TYPE_DEVICE  = CL_MEM_TYPE_DEVICE_INTEL,
+  PI_MEM_TYPE_SHARED  = CL_MEM_TYPE_SHARED_INTEL
+} _pi_usm_type;
+
+typedef enum : pi_bitfield  {
+  PI_MEM_ALLOC_FLAGS = CL_MEM_ALLOC_FLAGS_INTEL
+} _pi_usm_mem_properties;
+
+typedef enum : pi_bitfield {
+  PI_USM_MIGRATION_TBD0 = (1 << 0)
+} _pi_usm_migration_flags;
+
+typedef enum {
+  PI_INDIRECT_HOST_ACCESS   = CL_KERNEL_EXEC_INFO_INDIRECT_HOST_ACCESS_INTEL,
+  PI_INDIRECT_DEVICE_ACCESS = CL_KERNEL_EXEC_INFO_INDIRECT_DEVICE_ACCESS_INTEL,
+  PI_INDIRECT_SHARED_ACCESS = CL_KERNEL_EXEC_INFO_INDIRECT_SHARED_ACCESS_INTEL,
+  PI_USM_PTRS               = CL_KERNEL_EXEC_INFO_USM_PTRS_INTEL
+} _pi_usm_kernel_exec_info;
+
+typedef _pi_usm_capability_query  pi_usm_capability_query;
+typedef _pi_usm_capabilities      pi_usm_capabilities;
+typedef _pi_mem_info              pi_mem_info;
+typedef _pi_usm_type              pi_usm_type;
+typedef _pi_usm_mem_properties    pi_usm_mem_properties;
+typedef _pi_usm_kernel_exec_info  pi_usm_kernel_exec_info;
+typedef _pi_usm_migration_flags   pi_usm_migration_flags;
+
+pi_result piextUSMHostAlloc(
+  void **                 result_ptr,
+  pi_context              context,
+  pi_usm_mem_properties * properties,
+  size_t                  size,
+  pi_uint32               alignment);
+
+pi_result piextUSMDeviceAlloc(
+  void **                 result_ptr,
+  pi_context              context,
+  pi_device               device,
+  pi_usm_mem_properties * properties,
+  size_t                  size,
+  pi_uint32               alignment);
+
+pi_result piextUSMSharedAlloc(
+  void **                 result_ptr,
+  pi_context              context,
+  pi_device               device,
+  pi_usm_mem_properties * properties,
+  size_t                  size,
+  pi_uint32               alignment);
+
+pi_result piextUSMFree(
+  pi_context context,
+  void *     ptr);
+
+pi_result piextUSMKernelSetArgMemPointer(
+  pi_kernel    kernel,
+  pi_uint32    arg_index,
+  size_t       arg_size,
+  const void * arg_value);
+
+pi_result piextUSMKernelSetIndirectAccess(
+  pi_kernel kernel,
+  pi_queue queue);
+
+pi_result piextUSMEnqueueMemset(
+  pi_queue         queue,
+  void *           ptr,
+  pi_int32         value,
+  size_t           count,
+  pi_uint32        num_events_in_waitlist,
+  const pi_event * events_waitlist,
+  pi_event *       event);
+
+pi_result piextUSMEnqueueMemcpy(
+  pi_queue         queue,
+  pi_bool          blocking,
+  void *           dst_ptr,
+  const void *     src_ptr,
+  size_t           size,
+  pi_uint32        num_events_in_waitlist,
+  const pi_event * events_waitlist,
+  pi_event *       event);
+
+pi_result piextUSMEnqueuePrefetch(
+  pi_queue               queue,
+  const void *           ptr,
+  size_t                 size,
+  pi_usm_migration_flags flags,
+  pi_uint32              num_events_in_waitlist,
+  const pi_event *       events_waitlist,
+  pi_event *             event);
+
+pi_result piextUSMEnqueueMemAdvise(
```

Comment:
Please do add doxygen comments (at least something short initially) so that the new interfaces are clearer.
The exsiting PI interfaces are mostly taken from OpenCL so not having doc for them is kind of OK for now.
But we shall do that for them too.